cmd/notary-server/config.go

```diff
@@ -256,3 +260,45 @@ func parseServerConfig(configFilePath string, hRegister healthRegister) (context
 		ConsistentCacheControlConfig: consistentCache,
 	}, nil
 }
+
+func setupConfigReloadTrap(oldConfig *viper.Viper, configFile string, envPrefix string) {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, notary.NotarySupportedSignals...)
+	go func() {
+		for {
+			reloadConfig(<-c, configFile, oldConfig, envPrefix)
+		}
+	}()
+}
+
+// reloadConfig was designed to do all the dirty work that needed to reload
+// the config for notary-server and make the entrance-ReloadConfiguration()
+// for hot configuration reload do some generic things.
+func reloadConfig(sig os.Signal, configFile string, oldConfig *viper.Viper, envPrefix string) {
```

Comment:
This function doesn't seem to have any test coverage. @HuKeping could you add some tests for this?